### ISSUE TITLE
Original file name in `content-disposition` HTTP header

### ISSUE THREAD

**Issue by HT-7** [nokonoko/Uguu#104]
I suggest adding a `content-disposition` header that contains the original file name.

The `content-disposition` HTTP header does not work just for downloads, but also for inline files that are previewed in the browser. For example, YouTube channel banners have `content-disposition: inline; filename="channels4_banner.jpg"`, so that file name shows in the title and as a pre-filled file name in the "Save as" window.

**Comment by nokonoko** [comment]
This could possibly leak information not meant to be shared, however adding a option to enable this would work.

**Comment by HT-7** [comment]
Indeed. I recommend a checkbox on the uploader interface that reads "display file names publicly". It should be enabled by default.

![image](https://github.com/user-attachments/assets/bae131d9-e328-495c-bf9d-f519cd5b490e)

**Comment by nokonoko** [comment]
I'm gonna go ahead and close this issue, I still stand by my previous response about this being seen as sensitive information and very rarely something you want exposed. Having a checkbox for the user to decide does not follow in line how Uguu should function, as in "you just upload a file", when you start adding user defined options you quickly start to drift from that objective.

I might however add this as an option in the API in the future.
